pkg/controller/servingcert/starter/starter.go

```diff
@@ -139,3 +145,14 @@ func readIntermediateCACert(filename string) (*x509.Certificate, error) {
 	}
 	return certs[0], nil
 }
+
+// IsMicroshift checks if the operator is running on Microshift.
+func IsMicroshift(ctx context.Context, kubeClient *kubernetes.Clientset) (bool, error) {
+	if _, err := kubeClient.CoreV1().ConfigMaps("kube-public").Get(ctx, "microshift-version", metav1.GetOptions{}); err != nil {
```

Comment:
Is this really the proper way to determine if a cluster is a microshift cluster?